### DISCUSSION TITLE
Gas estimation

### DISCUSSION THREAD

**Discussion by arlgnv** [wevm/wagmi#808] [Q&A]
Can I get gas estimation from `usePrepareContractWrite` hook? I need it to display to user and to calculate total amount.

upd. I think I found it. Is this info stored in `config.request.gasLimit` field in response from `usePrepareContractWrite` hook?
upd. No, I am not sure what it actually is 'cause MetaMask shows me different number for "gas estimation"

**Comment by dev-cz** [comment]
Also would like to see this. I gas estimation for the specific contract function to return would be awesome vs just the current gas price.

**Comment by martinang2** [comment]
Hey @alagunoff Any updates on this? I've been using the `config.request.gasLimit` as well but this differs greatly from MetaMask's number

**Reply by arlgnv** [reply]
> Hey @alagunoff Any updates on this? I've been using the `config.request.gasLimit` as well but this differs greatly from MetaMask's number

Nope, I haven't found a solution. I left that task and didn't encounter this kinda task anymore. You can try @challet's solution and if it helps let us know plz so I mark it as an answer.

**Comment by challet** [comment]
I don't know about the React hook, but the core [prepareWriteContract](https://wagmi.sh/core/actions/prepareWriteContract), on which it is probably based, is making a gas estimate. It should be available in its return value `request.data.gasLimit`.

Update : Or you can directly call the ethers's [estimateGas](https://docs.ethers.org/v5/api/contract/contract/#contract-estimateGas) method if you have a contract instance.

**Comment by StKostyantin** [comment]
gas price(in gwei) = gas limit * (base fee + priority fee)
gwei to ETH = gwei / 10**9
ETH*currency = price fee

data.gasPrice.standard - get get from [etherscan gastracker](https://etherscan.io/gastracker)

const ethGasPrice = config?.request?.gasLimit.mul(data.gasPrice.standard).toNumber() / 10 ** 9;
const usdGasPrice = ethGasPrice * 1300

[What Is Ethereum Gas](https://www.sofi.com/learn/content/what-is-ethereum-gas/)
